tests/unit/models/transactions/test_deposit_preauth.py

```diff
@@ -2,27 +2,163 @@
+                        + "."
+                        + " It is "
+                        + "invalid if none of the params are specified."
+                        + "'}",
+                    )
+                # capture the case where a plurality of inputs are specified
+                else:
+                    self.assertEqual(
+                        error.exception.args[0],
+                        "{'DepositPreauth': '"
+                        + "More than one input param cannot be specified for "
+                        + "DepositPreauth "
+                        + "transaction. Please specify exactly one input parameter "
+                        + "amongst "
+                        + "authorize, unauthorize, authorize_credentials and "
+                        + "unauthorize_credentials."
+                        + "'}",
+                    )
+
+    # Unit tests validating the length of array inputs
+    def test_authcreds_array_input_exceed_length_check(self):
+        # Note: If credentials de-duplication is implemented in the client library,
+        # additional tests need to be written
+        sample_credentials = [
+            Credential(
+                issuer="SampleIssuer_" + str(i), credential_type="SampleCredType"
+            )
+            for i in range(9)
+        ]
+
+        with self.assertRaises(XRPLModelException) as error:
             DepositPreauth(
                 account=_ACCOUNT,
                 fee=_FEE,
                 sequence=_SEQUENCE,
-                authorize="authorize",
-                unauthorize="unauthorize",
+                authorize_credentials=sample_credentials,
             )
 
-    def test_authorize_unauthorize_neither_set(self):
-        with self.assertRaises(XRPLModelException):
+        self.assertEqual(
+            error.exception.args[0],
+            "{'DepositPreauth_authorize_credentials': '"
+            + "AuthorizeCredentials list cannot have more than 8 elements. "
+            + "'}",
+        )
+
+    def test_authcreds_empty_array_inputs(self):
+        with self.assertRaises(XRPLModelException) as error:
             DepositPreauth(
                 account=_ACCOUNT,
                 fee=_FEE,
                 sequence=_SEQUENCE,
+                authorize_credentials=[],
             )
+
+        self.assertEqual(
+            error.exception.args[0],
+            "{'DepositPreauth_authorize_credentials': '"
+            + "AuthorizeCredentials list cannot be empty. "
+            + "'}",
+        )
+
+    def test_unauthcreds_array_input_exceed_length_check(self):
+        # Note: If credentials de-duplication is implemented in the client library,
+        # additional tests need to be written
+        sample_credentials = [
+            Credential(
+                issuer="SampleIssuer_" + str(i), credential_type="SampleCredType"
+            )
+            for i in range(9)
+        ]
```

Comment:
```suggestion
        sample_credentials = [
            Credential(
                issuer="SampleIssuer_" + str(i), credential_type="SampleCredType"
            )
        ]*9
```

Reply:
I'm not able to make use of the `i` counter variable in your suggested syntax

Reply:
Ah, good point.